state/cleanup_test.go

```diff
@@ -614,6 +614,33 @@ func (s *CleanupSuite) TestForceDestroyMachineSchedulesRemove(c *gc.C) {
 	c.Assert(err, jc.Satisfies, errors.IsNotFound)
 }
 
+func (s *CleanupSuite) TestRemoveApplicationRemovesAllCleanUps(c *gc.C) {
```

Comment:
Can we enhance the test to also add an application and units for say "dummyfoo" to ensure they are not accidentally removed.